solarforecastarbiter/datamodel.py

```diff
@@ -914,25 +914,47 @@ class ForecastObservation(BaseModel):
     forecast: :py:class:`solarforecastarbiter.datamodel.Forecast`
     observation: :py:class:`solarforecastarbiter.datamodel.Observation`
     reference_forecast: :py:class:`solarforecastarbiter.datamodel.Forecast` or None
-    normalization: float
+    normalization: float or None
+        If None, determined by __set_normalization__
     cost_per_unit_error: float
     """  # NOQA
     forecast: Forecast
     observation: Observation
     reference_forecast: Union[Forecast, None] = None
-    # for now, some function applied to observation (e.g. mean per day)
-    # possible in future
-    normalization: float = 1.0
+    # some function applied to observation (e.g. mean per day)
+    # possible in future. maybe add pd.Series like for
+    # ProcessedForecastObservation
+    normalization: Union[float, None] = None
     # cost: Cost
     cost_per_unit_error: float = 0.0
     data_object: Observation = field(init=False)
 
     def __post_init__(self):
+        if self.normalization is None:
+            __set_normalization__(self)
```

Comment:
seems like the current pattern would be to pass self.observation instead of self

Reply:
I went back and forth on this. I ultimately went with `self` since `__set_units__` takes that (well, `cls`) and then calls `__set_attr__`. `__set_constant_values__` also takes `self`, but it's more complicated.  Which functions are you looking at?

Reply:
ah did forget about setting the attribute, so that's fine